### PR TITLE
Fix the metadata flags test flake

### PR DESCRIPTION
**Pull request by lidizheng**
Explicitly close the socket in the test case to ensure the connectivity state transition from CONNECTING into TRANSIENT FAILURE.

The root cause might originated from a behavior change in Core. A bisect might be able to find it, but probably not worth it (the change could be legit for other cases). This PR should fix it within Python wrapper.

Fixes https://github.com/grpc/grpc/issues/22760

### CODE REVIEW

File: src/python/grpcio_tests/tests/unit/_metadata_flags_test.py [hunk 1]
```diff
@@ -101,8 +102,9 @@ def service(self, handler_call_details):
 
 def create_dummy_channel():
     """Creating dummy channels is a workaround for retries"""
-    with bound_socket() as (host, port):
-        return grpc.insecure_channel('{}:{}'.format(host, port))
+    host, port, sock = get_socket()
+    sock.close()
```
Comment (gnossen) [inline src/python/grpcio_tests/tests/unit/_metadata_flags_test.py:106]: Without `SO_REUSEPORT`, there's a race condition here, right? The thread gets preempted between `sock.close()` and `grpc.insecure_channel`, and then the kernel allocates `port` to some other process on the system. Then `grpc.insecure_channel` fails to bind. I'm not sure what the failure rate should be though. Can you run with `--runs_per_test=10000` to find out whether or not this is worth worrying about?

File: src/python/grpcio_tests/tests/unit/_metadata_flags_test.py [hunk 2]
```diff
@@ -203,51 +205,56 @@ def test_call_wait_for_ready_enabled(self):
...
         unhandled_exceptions = queue.Queue()
-        with bound_socket(listen=False) as (host, port):
-            addr = '{}:{}'.format(host, port)
-            wg = test_common.WaitGroup(len(_ALL_CALL_CASES))
 
-            def wait_for_transient_failure(channel_connectivity):
-                if channel_connectivity == grpc.ChannelConnectivity.TRANSIENT_FAILURE:
+        # We just need an unused TCP port
+        host, port, sock = get_socket()
+        sock.close()
```
Comment (gnossen) [inline src/python/grpcio_tests/tests/unit/_metadata_flags_test.py:211]: Same comment as above. It looks like there's a race condition here.

**Comment by lidizheng** [comment]
```
INFO: Elapsed time: 5980.388s, Critical Path: 68.72s
INFO: 40006 processes: 40006 linux-sandbox.
INFO: Build completed, 2 tests FAILED, 20007 total actions
//src/python/grpcio_tests/tests/unit:_metadata_flags_test.python2       TIMEOUT in 6 out of 10000 in 60.1s
//src/python/grpcio_tests/tests/unit:_metadata_flags_test.python3       TIMEOUT in 3 out of 10000 in 60.1s
  Stats over 10000 runs: max = 60.1s, min = 3.0s, avg = 3.5s, dev = 1.0s
```

Flake rate is around 0.045% (9/20000). @gnossen PTALA.

**Comment by lidizheng** [comment]
Known failures: https://github.com/grpc/grpc/issues/22875 https://github.com/grpc/grpc/issues/22691

**Pull request merged**